### PR TITLE
Performance improvements.

### PR DESCRIPTION
**Pull request by splatteredbits**
Debug output to help users track down/understand their glob patterns. 

Allow searching hidden files/folders.

### CODE REVIEW

File: Glob/Functions/Find-GlobFile.ps1 [hunk 1]
```diff
@@ -46,84 +59,169 @@ function Find-GlobFile
...
         try
         {
-            foreach( $fileInfo in (Get-ChildItem -Path $rootPath -File -Recurse) )
+            if( -not $IsDirectory )
             {
-                $relativePath = 
-                    $fileInfo.FullName | 
-                    Resolve-Path -Relative | 
-                    ForEach-Object { $_ -replace '^\.(\\|/)','' }
                 $matches = $false
```
Comment (decoyjoe) [inline Glob/Functions/Find-GlobFile.ps1]: `$Matches` is an [automatic variable](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-5.1#matches) related to the `-match` operator.

Use a different name for this variable?

File: Glob/Functions/Find-GlobFile.ps1 [hunk 2]
```diff
@@ -46,84 +59,169 @@ function Find-GlobFile
...
 
                 if( -not $matches )
                 {
-                    continue
+                    return $false
                 }
+            }
 
-                foreach( $excludeGlob in $excludeGlobs )
+            $matches = $true
```
Comment (decoyjoe) [inline Glob/Functions/Find-GlobFile.ps1]: `$Matches` is an [automatic variable](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-5.1#matches) related to the `-match` operator.

Use a different name for this variable?

File: Glob/Glob.psd1 [hunk 3]
```diff
@@ -110,8 +110,11 @@
 
             # ReleaseNotes of this module
             ReleaseNotes = '
-* Fixed: Find-GlobFile doesn''t work on Linux/macOS.
-* Fixed: Find-GlobFile doesn''t respect user''s preference variables.
+* Fixed: `Find-GlobFile` doesn''t work on Linux/macOS.
+* Fixed: `Find-GlobFile` doesn''t respect user''s preference variables.
+* Improved performance. If a directory matches an excluded pattern, Glob no longer recurses into it to find files that are just going to be excluded.
```
Comment (decoyjoe) [inline Glob/Glob.psd1:115]: No test was added for skipping files under excluded _directories_.

**Review by decoyjoe — CHANGES_REQUESTED** [review]

File: Tests/Find-GlobFile.Tests.ps1 [hunk 4]
```diff
@@ -291,4 +298,21 @@ else
...
+if( $IsWindows )
+{
+    Describe 'Find-GlobFile.when excluding a directory' {
+        It 'should not recurse into excluded directories' {
+            Init
+            GivenFile 'file.txt','excluded\file.txt'
+            # Create a circular junction. When you recurse too deeply, eventually PowerShell throws an error that you've
+            # gone too deep. If it got excluded properly, there won't be an error.
+            New-CJunction -Link (Join-Path -Path $testRoot -ChildPath 'excluded\circular') `
+                          -Target (Join-Path -Path $testRoot -ChildPath 'excluded')
```
Comment (decoyjoe) [inline Tests/Find-GlobFile.Tests.ps1:312]: Does this still apply if a system has [NTFS long paths](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later) enabled?

**Review by decoyjoe — APPROVED** [review]

File: Tests/Find-GlobFile.Tests.ps1 (hunk 4 above)
Comment (splatteredbits) [reply Tests/Find-GlobFile.Tests.ps1:312]: 🤷‍♀  If they do, then the test will just run and run and run and run. Maybe eventually Windows will run out of memory and crash?

**Pull request merged**